### PR TITLE
Fix regressor cost

### PR DESCRIPTION
**Pull request by skearnes**
The regression loss function is wrong in tfvs. `tf.nn.l2_loss` returns a scalar (the unweighted sum of the per-example losses), which obviously gives the wrong values for the per-example losses. This PR calculates per-example losses which are then weighted appropriately.

### CODE REVIEW

**Comment by rbharath** [comment]
Thanks for the fix!
